### DISCUSSION TITLE
libusb port does not target WASM/emscripten under the wasm32-emscripten triplet

### DISCUSSION THREAD

**Discussion by t1-nva** [microsoft/vcpkg#35289] [Q&A] [answered]
When trying to build libusb with the wasm32-emscripten triplet the host os version is attempted, not the emscripten/WASM libusb backend. Yes, there really is a libusb backend for webassembly/emscripten :-)

I attempted to fix this in the port files, but was unsuccessful both on linux/WSL2 and windows.

I did succeed building it standalone according to the build instructions here on linux though; https://web.dev/articles/porting-libusb-to-webusb#using_the_port, but that includes using the emscripten-supplied emconfigure and emmake, which I'm not sure how or if I should use in the vcpkg port.

Any hints to how this could be fixed would be appreciated

**Comment by dg0yt** [comment, marked as answer]
You probably don't need `emconfigure`/`emmake`, but `--host=wasm32` isn't there yet. Test #35292. (I tested the build, but I don't know if it is usable.)
